src/modules/export.js

```diff
@@ -342,20 +342,43 @@ export const sampleFields = () => {
       ? { filter: {} }
       : exportData.query;
 
-    dataService.find(ns, spec.filter, {limit: 1}, function(findErr, docs) {
-      if (findErr) {
-        return onError(findErr);
-      }
-
-      // Use `dotnotation.serialize()` to recurse into documents and
-      // pick up all possible paths.
-      const fields = Object.keys(dotnotation.serialize(docs[0])).sort().reduce((obj, field) => {
-        obj[field] = 1;
+    const sampleOptions = {
+      query: spec.filter,
+      size: 100,
+      promoteValues: true
+    };
 
-        return obj;
-      }, {});
+    // To make sure we have less errors in the fields we generate for export, we
+    // are sampling 100 documents from the current collection and then sending
+    // those documents through mongodb-schema.
+    const samplingStream = dataService.sample(ns, sampleOptions);
+    const analyzingStream = schemaStream();
+
+    // Grab all the field paths that mongodb-schema generates for us.
+    const parseSchema = new stream.Transform({
+      writableObjectMode: true,
+      transform(chunk, encoding, callback) {
+        const fields = chunk.fields.reduce((obj, field)=> {
```

Comment:
This manipulation could have been 'smarter'. I chose to spell it out a bit more, since schema stuff can be so confusing if you don't look at it for a couple of months.